--- net/netfilter/xt_qtaguid.c
+++ net/netfilter/xt_qtaguid.c
@@ -827,4 +827,4 @@	int fmt = (int)data; /* The data is just 1 (old) or 2 (uses fmt) */
	struct iface_stat *iface_entry;
-	struct rtnl_link_stats64 dev_stats, *stats;
+	struct rtnl_link_stats64 *stats;
 struct rtnl_link_stats64 no_dev_stats = {0};
 
 
@@ -864,13 +864,8 @@	list_for_each_entry(iface_entry, &iface_stat_list, list) {
 		if (item_index++ < items_to_skip)
			continue;
 
-	if (iface_entry->active) {
-		stats = dev_get_stats(iface_entry->net_dev,
-				      &dev_stats);
-	} else {
-		stats = &no_dev_stats;
-	}
+	stats = &no_dev_stats;
 	/*
 	 * If the meaning of the data changes, then update the fmtX
 	 * string.
